gtsfm/loader/astronet_loader.py

```diff
@@ -0,0 +1,242 @@
+
+
+        Args:
+            data_dir: path to directory containing the COLMAP-formatted data: cameras.bin, images.bin, and points3D.bin
+            gt_scene_mesh_path (optional): path to file of target small body surface mesh.
+                Note: vertex size mismath observed when reading in from OBJ format. Prefer PLY.
+            use_gt_intrinsics: whether to use ground truth intrinsics. If calibration is
+               not found on disk, then use_gt_intrinsics will be set to false automatically.
+            use_gt_extrinsics (optional): whether to use ground truth extrinsics
+            use_gt_sfmtracks (optional): whether to use ground truth tracks
+            max_frame_lookahead (optional): maximum number of consecutive frames to consider for
+                matching/co-visibility. Any value of max_frame_lookahead less than the size of
+                the dataset assumes data is sequentially captured
+
+        Raises:
+            FileNotFoundError if image path does not exist
+        """
+        self._use_gt_intrinsics = use_gt_intrinsics
+        self._use_gt_extrinsics = use_gt_extrinsics
+        self._use_gt_sfmtracks = use_gt_sfmtracks
+        self._max_frame_lookahead = max_frame_lookahead
+
+        # Use COLMAP model reader to load data and convert to GTSfM format
+        if Path(data_dir).exists():
+            _cameras, _images, _points = read_model(path=data_dir, ext=".bin")
+            self._calibrations, self._wTi_list, img_fnames, self._sfmtracks = self.colmap2gtsfm(
+                _cameras, _images, _points, load_sfmtracks=use_gt_sfmtracks
+            )
+
+        # Read in scene mesh as Trimesh object
+        if gt_scene_mesh_path is not None:
+            if not Path(gt_scene_mesh_path).exists():
+                raise FileNotFoundError(f"No mesh found at {gt_scene_mesh_path}")
+            self._gt_scene_trimesh = trimesh.load(gt_scene_mesh_path, process=False, maintain_order=True)
+        else:
+            self._gt_scene_trimesh = None
+
+        if self._calibrations is None:
+            self._use_gt_intrinsics = False
+
+        if self._wTi_list is None:
+            self._use_gt_extrinsics = False
+
+        if self._sfmtracks is None:
+            self._use_gt_sfmtracks = False
+        self.num_sfmtracks = len(self._sfmtracks) if self._sfmtracks is not None else 0
+
+        # Prepare image paths
+        self._image_paths = []
+        for img_fname in img_fnames:
+            img_fpath = os.path.join(data_dir, "images", img_fname)
+            if not Path(img_fpath).exists():
+                raise FileNotFoundError(f"Could not locate image at {img_fpath}.")
+            self._image_paths.append(img_fpath)
+
+        self._num_imgs = len(self._image_paths)
+        logger.info("AstroNet loader found and loaded %d images and %d tracks.", self._num_imgs, self.num_sfmtracks)
+
+    @staticmethod
+    def colmap2gtsfm(
```

Comment:
this seems like this is a more general purpose function that should go into GtsfmData or something, like 

```python
class GtsfmData:
     @classmethod
     def from_colmap_model(cls, colmap_model_dir: str) -> "GtsfmData":
```
would that make sense?

Reply:
Yeah that's a good idea; I'll do that.

Reply:
@johnwlambert I think this may be better for another PR, as I think the `GtsfmData` calss needs to be restructured slightly for the loaders to be able to utilize it (e.g., storing image names for reading).

Reply:
Moved this to an issue, as I think it's out of the scope of this PR.

Reply:
I don't think this function belongs in this file -- I'd prefer we avoid the tech debt and move it elsewhere before we check it in.

Reply:
I think every loader should have a `to_gtsfmdata()` method that converts the respective data formats to `GtsfmData`. I think putting all of these data conversion methods into the core code for every dataset we use will just clutter things. I think the loader should be more self contained, so that if someone wants to implement GTSfM on their data, they just need to fill in the loader.

Maybe the `LoaderBase` should have an abstract method. Maybe it would look something like this:

```python
@abc.abstractmethod
def to_gtsfmdata(self, **kwargs: Dict[str, Any]) -> GtsfmData:
    """Converts individual data formats to `GtsfmData`."""
```

I think leaving this here untill we address [#263](https://github.com/borglab/gtsfm/issues/263) makes sense.